src/lib/registration/widgets/package_search.rb

```diff
@@ -0,0 +1,215 @@
+    class PackageSearch < CWM::CustomWidget
+      include Yast::Logger
+
+      # @return [Array<String>] List of selected packages
+      attr_reader :selected_packages
+
+      # @return [::Registration::PackageSearch,nil] Current search
+      attr_reader :search
+
+      # Constructor
+      def initialize
+        textdomain "registration"
+        self.handle_all_events = true
+        @selected_packages = [] # list of selected packages
+        super
+      end
+
+      # @macro seeAbstractWidget
+      def contents
+        MarginBox(
+          0.5,
+          0.5,
+          HBox(
+            VBox(
+              search_form,
+              VStretch()
+            ),
+            MinWidth(
+              60,
+              VBox(
+                packages_table,
+                package_details
+              )
+            )
+          )
+        )
+      end
+
+      # @macro seeAbstractWidget
+      def handle(event)
+        if start_search_event?(event)
+          search_package(search_form.text)
+        elsif event["WidgetID"] == "remote_packages_table"
+          handle_packages_table_event(event)
+        end
+
+        log.debug "Event handled #{event.inspect}"
+        nil
+      end
+
+    private
+
+      # Search form widget
+      #
+      # @return [PackageSearchForm] Search form widget instance
+      def search_form
+        @search_form ||= PackageSearchForm.new
+      end
+
+      # Packages table widget
+      #
+      # This widget is used to display the result of the search.
+      #
+      # @return [RemotePackagesTable] Packages table widget
+      def packages_table
+        @packages_table ||= RemotePackagesTable.new
+      end
+
+      # Package details widget
+      #
+      # This widget displays the details of the package which is selected in the
+      # table.
+      #
+      # @return [RemotePackageDetails] Package details widget.
+      def package_details
+        @package_details ||= RemotePackageDetails.new
+      end
+
+      # Handles remote packages table events
+      #
+      # @param event [Hash] Widget event to process
+      def handle_packages_table_event(event)
+        case event["EventReason"]
+        when "Activated"
+          toggle_package
+        when "SelectionChanged"
+          update_details
+        end
+      end
+
+      # Determines whether the event should start the search
+      #
+      # @param event [Hash] UI event to check
+      # @return [Boolean]
+      def start_search_event?(event)
+        event["WidgetID"] == "search_form_button" ||
+          (event["WidgetID"] == "search_form_text" && event["EventReason"] == "Activated")
+      end
+
+      # Performs the search and updates the packages table
+      #
+      # @param text [String] Text to search for
+      def search_package(text)
+        @search = ::Registration::PackageSearch.new(text: text)
+        Yast::Popup.Feedback(_("Searching..."), _("Searching for packages with the given name")) do
```

Comment:
`TRANSLATORS:` comments?

BTW I would use something more generic instead of the `with the given name` part. Just in case we later allow searching in summary or package description...